feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch

```diff
@@ -0,0 +1,1489 @@
++            r->n_event_report * sizeof(Sts__EventReport*));
++
++    // allocate new buffer
++    sr = malloc(sizeof(Sts__EventReport));
++    size += sizeof(Sts__EventReport);
++    assert(sr);
++    r->event_report[r->n_event_report - 1] = sr;
++    sts__event_report__init(sr);
++    //sr->timestamp_ms = events->timestamp_ms;
++
++    for (i = 0; i < events->qty; i++)
++    {
++        // dpp event pointer
++        dpp_event_record_t *rec = &events->list[i];
++
++        // ClientAssocEvent
++        if(rec->event_type == ET_CLIENT_ASSOC){
++            // init
++            sts__event_report__client_assoc_event__init(dr_assoc);
++            // STA mac
++            dr_assoc->sta_mac = malloc(MACADDR_STR_LEN);
++            dpp_mac_to_str(rec->client_assoc_event.sta_mac, dr_assoc->sta_mac);
++            // SSID
++            if(rec->client_assoc_event.ssid)
++                dr_assoc->ssid = strcpy(dr_assoc->ssid, rec->client_assoc_event.ssid);
++
++        // ClientAuthEvent
++        }else if (rec->event_type == ET_CLIENT_AUTH){
++            // init
++            sts__event_report__client_auth_event__init(dr_auth);
++            // STA mac
++            dr_auth->sta_mac = malloc(MACADDR_STR_LEN);
++            dpp_mac_to_str(rec->client_auth_event.sta_mac, dr_auth->sta_mac);
++            // SSID
++            if(rec->client_auth_event.ssid)
++                dr_auth->ssid = strcpy(dr_auth->ssid, rec->client_auth_event.ssid);
++
++        // ClientDisconnectEvent
++        }else if (rec->event_type == ET_CLIENT_DISCONNECT){
++            // init
++            sts__event_report__client_disconnect_event__init(dr_dscn);
++            // STA mac
++            dr_dscn->sta_mac = malloc(MACADDR_STR_LEN);
++            dpp_mac_to_str(rec->client_disconnect_event.sta_mac, dr_dscn->sta_mac);
++            // SSID
++            if(rec->client_disconnect_event.ssid)
++                dr_dscn->ssid = strcpy(dr_dscn->ssid, rec->client_disconnect_event.ssid);
++
++        }
++    }
++}
++
+ static void dppline_add_stat_device(Sts__Report *r, dppline_stats_t *s)
+ {
+     Sts__Device *sr = NULL;
+@@ -1942,6 +2166,10 @@ static void dppline_add_stat(Sts__Report
+             dppline_add_stat_network_probe(r, s);
+             break;
+ 
++        case DPP_T_EVENTS:
++            dppline_add_stat_events(r, s);
++            break;
++
+         default:
+             LOG(ERR, "Failed to add %d to stats report", s->type);
+             /* do nothing       */
+@@ -2106,6 +2334,14 @@ bool dpp_put_network_probe(dpp_network_p
+ }
+ 
+ /*
++ * Put event report to internal queue
++ */
++bool dpp_put_events(dpp_event_report_data_t *rpt)
++{
++    return dppline_put(DPP_T_EVENTS, rpt);
++}
++
++/*
+  * Create the protobuf buff and copy it to given buffer
+  */
+ #ifndef DPP_FAST_PACK
+Index: opensync-2.0.5.0/platform/openwrt/src/lib/target/inc/target_OPENWRT.h
+===================================================================
```

Comment:
dont patches files in the src/ folder but apply the patch to the actual file please

Reply:
What do you mean by this exactly? All the other patches (**16-dhcp-client.patch** for example) do the same.
```
Index: opensync-2.0.5.0/src/lib/datapipeline/inc/dpp_network_probe.h
===================================================================
--- opensync-2.0.5.0.orig/src/lib/datapipeline/inc/dpp_network_probe.h
+++ opensync-2.0.5.0/src/lib/datapipeline/inc/dpp_network_probe.h
```

Reply:
I figured out what you were saying. Closing this.